pycqed/measurement/waveform_control/circuit_builder.py

```diff
@@ -335,4 +335,19 @@ def seq_from_ops(self, operations, fill_values=None,  pulse_modifs=None,
                                       pulse_modifs=pulse_modifs).build()
         pulses += self.mux_readout(**ro_kwargs).build()
         seq.add(Segment('Segment1', pulses))
-        return seq
\ No newline at end of file
+        return seq
+
+    def simultaneous_blocks(self, block_name, blocks):
+        simultaneous = Block(block_name, [])
+        simultaneous_end_pulses = []
+        for block in blocks:
+            simultaneous.extend(block.build(ref_pulse=f"start"))
+            simultaneous_end_pulses.append(simultaneous.pulses[-1]['name'])
+        simultaneous.extend([{"name": f"simultaneous_end_pulse",
```

Comment:
Do you have to do this in this way because you want to return the non compiled block but still adjust references already now and not when compiling? In the future we might have to think of a better structure because here we "duplicate" the used of the "end pulse" that we usually add when compiling...

Reply:
Yes, this is the reason. We should keep this in mind at the point where we more generally discuss about (pre-)compilation of blocks. The current version of the multiref code and of this functions were written in a way that we do not have to change anything in the general structure of build() and resolve_...()